v2/clients/http/command.go

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright (C) 2021 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"context"
+	"net/url"
+	"path"
+	"strconv"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/clients/http/utils"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/clients/interfaces"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/dtos/responses"
+)
+
+type CommandClient struct {
+	baseUrl string
+}
+
+// NewCommandClient creates an instance of CommandClient
+func NewCommandClient(baseUrl string) interfaces.CommandClient {
+	return &CommandClient{
+		baseUrl: baseUrl,
+	}
+}
+
+// AllDeviceCoreCommands returns a paginated list of MultiDeviceCoreCommandsResponse. The list contains all of the commands in the system associated with their respective device.
+func (client *CommandClient) AllDeviceCoreCommands(ctx context.Context, offset int, limit int) (
+	res responses.MultiDeviceCoreCommandsResponse, err errors.EdgeX) {
+	requestParams := url.Values{}
+	requestParams.Set(v2.Offset, strconv.Itoa(offset))
+	requestParams.Set(v2.Limit, strconv.Itoa(limit))
+	err = utils.GetRequest(ctx, &res, client.baseUrl, v2.ApiAllDeviceRoute, requestParams)
+	if err != nil {
+		return res, errors.NewCommonEdgeXWrapper(err)
+	}
+	return res, nil
+}
+
+// DeviceCoreCommandsByDeviceName returns all commands associated with the specified device name.
```

Comment:
Why is name `DeviceCoreCommandsByDeviceName` rather than just `CoreCommandsByDeviceName`?

Reply:
Since we have DeviceCoreCommand and CoreCommand DTO.  See https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/v2/dtos/corecommand.go

And the device profile structure is
```
name: "Simple-Device"
   ...
deviceResources:
   ...
deviceCommands:
   ...
```
And the core-command API looks like
```
% curl http://localhost:48082/api/v2/device/all | json_pp
{
   "apiVersion" : "v2",
   "deviceCoreCommands" : [
      {
         "coreCommands" : [
            {
               "get" : true,
               "name" : "Switch",
               "parameters" : [
                  {
                     "resourceName" : "SwitchButton",
                     "valueType" : "Bool"
                  }
               ],
               "path" : "/api/v2/device/name/Simple-Device01/Switch",
               "set" : true,
               "url" : "http://localhost:48082"
            },
            ...
            ...
         ],
         "deviceName" : "Simple-Device01",
         "profileName" : "Simple-Device"
      }
   ],
   "statusCode" : 200
}
```

So the DeviceCoreCommand is fine to me. Do you think the `CoreCommandsByDeviceName` is more appropriate or we should rename the model or dto?